c/planarityTestGraphFunctionality.c

```diff
@@ -140,36 +160,100 @@ int TestGraphFunctionality(char *commandString, char *infileName, char *inputStr
 					char *finalSlash = strrchr(infileName, FILE_DELIMITER);
 					char *infileBasename = finalSlash ? (finalSlash + 1) : infileName;
 
-					char *headerStr = (char *) malloc((strlen(infileBasename) + 3) * sizeof(char));
+					char *headerFormat = "FILENAME=\"%s\" DURATION=\"%.3lf\"\n";
+					char *headerStr = (char *) malloc(
+														(
+															strlen(headerFormat) +
+															strlen(infileBasename) +
+															strlen("-1.7976931348623158e+308") + // -DBL_MAX from float.h
+															3
+														) * sizeof(char));
 					if (headerStr == NULL)
 					{
 						ErrorMessage("Unable allocate memory for output file header.\n");
+
 						free(inputStr);
 						inputStr = NULL;
+
 						gp_Free(&theGraph);
+
 						sf_Free(&testOutput);
+
 						return NOTOK;
 					}
 
-					sprintf(headerStr, "%s\n", infileBasename);
+					testAllStatsP stats = (testAllStatsP) calloc(1, sizeof(testAllStats));
+					if (stats == NULL)
+					{
+						ErrorMessage("Unable allocate memory for test stats.\n");
+
+						free(inputStr);
+						inputStr = NULL;
+
+						gp_Free(&theGraph);
+
+						sf_Free(&testOutput);
+
+						free(headerStr);
+						headerStr = NULL;
+
+						return NOTOK;
+					}
+
+					char command = commandString[1];
+					Result = testAllGraphs(theGraph, command, inputStr, stats);
+
+					// Stop the timer
+					platform_GetTime(end);
+					duration = platform_GetDuration(start, end);
+					sprintf(messageContents, "\nDone testing all graphs (%.3lf seconds).\n", duration);
+					Message(messageContents);
+
+					sprintf(headerStr, headerFormat, infileBasename, duration);
```

Comment:
Call the output function here. Be careful to preserve a NOTOK result from `testAllGraphs()`. In other words, call as `if (myOutputFunc(outfileName, stats) != OK) Result = NOTOK;` rather than just `Result = myOutputFunc(...)`.

Reply:
Whoops, almost forgot this one! Addressed in latest commit.